src/vl-footer.js

```diff
@@ -70,7 +76,45 @@ export class VlFooter extends vlElement(HTMLElement) {
     if (!VlFooter.footer) {
       document.body.appendChild(this.getFooterTemplate());
     }
-    eval(code.replace(/document\.write\((.*?)\);/, 'document.getElementById("' + VlFooter.id + '").innerHTML = $1;'));
+    this.__observeFooterElementIsAdded();
+    Function(code.replace(/document\.write\((.*?)\);/, 'document.getElementById("' + VlFooter.id + '").innerHTML = $1;'))();
+  }
+
+  __observeFooterElementIsAdded() {
+    const target = document.querySelector('#' + VlFooter.id);
+    const footerObserver = new MutationObserver((mutations, observer) => this.__footerObserverCallback(mutations, observer));
+    footerObserver.observe(target, {childList: true});
+  }
+
+  __footerObserverCallback(mutations, observer) {
+    if (this.__footerIsToegevoegdIneenVanDeMutaties(mutations)) {
+      this.dispatchEvent(new CustomEvent(VlFooter.EVENTS.ready));
+      observer.disconnect();
+    }
+  }
+
+  __footerIsToegevoegdIneenVanDeMutaties(mutations) {
+    return mutations.some((mutation) => {
+      return mutation.type === 'childList' && this.__footerElementIsToegevoegd(mutation.addedNodes);
+    });
+  }
+
+  __footerElementIsToegevoegd(toegevoegdeNodes) {
+    return this.__eenVanDeNodesBevatElement(toegevoegdeNodes, 'FOOTER');
+  }
+
+  __eenVanDeNodesBevatElement(nodeList, element) {
+    return nodeList && this.__nodeListBevatNodeMetElement(nodeList, element);
+  }
+
+  __nodeListBevatNodeMetElement(nodeList, element) {
+    return Array.from(nodeList).some((node) => {
+      return this.__nodeIsElementOfHeeftElementAlsChild(node, element);
+    });
+  }
+
+  __nodeIsElementOfHeeftElementAlsChild(node, element) {
+    return node.tagName === element || this.__eenVanDeNodesBevatElement(node.childNodes, element);
   }
```

Comment:
Overdaad aan private functies die alleen maar relevant zijn voor de mutation observer waardoor de leesbaarheid moeilijk wordt en er steeds variabelen doorgegeven moeten worden.

Reply:
Waar krijgt this._observer een waarde?

Reply:
In de `__executeCode` al werd er geen return statement uitgevoerd, heb ik nu nog toegevoegd.